### ISSUE TITLE
Build fails on arm64: error: static assertion failed due to requirement 'is_supported_integral_type<char>'

### ISSUE THREAD

**Issue by yurivict** [facebook/fbthrift#623]
```
In file included from /wrkdirs/usr/ports/devel/fbthrift/work/fbthrift-2024.09.23.00/thrift/compiler/whisker/eval_context.cc:17:
In file included from /wrkdirs/usr/ports/devel/fbthrift/work/fbthrift-2024.09.23.00/./thrift/compiler/whisker/eval_context.h:20:
/wrkdirs/usr/ports/devel/fbthrift/work/fbthrift-2024.09.23.00/./thrift/compiler/whisker/object.h:515:15: error: static assertion failed due to requirement 'is_supported_integral_type<char>'
  515 | static_assert(is_supported_integral_type<char>);
      |               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.
```

[log](https://pkg-status.freebsd.org/ampere3/data/141arm64-default/1f84c1fae602/logs/fbthrift-2024.09.23.00.log)

Version: 2024.09.23.00
clang-18
FreeBSD 14.1

**Comment by praihan** [comment]
Ah `char` is unsigned on your platform. Seems like a simple fix. Let me see.

**Comment by praihan** [comment]
Fixed by c8c7192ab70a99906e076c143165ebe1536735ee.
